person_class.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknow', parent_permission: true)
+    @id_counter = 0
+    @id = @id_counter += 1
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```

Comment:
- [ ] Ouch! this is a close one but the id generated in the person class will not be unique for each object. The reason is that the `@id_counter` variable is initialized and incremented within the constructor. Every time an instance of the Person class is created, the initialize method will be called, and the `@id_counter` will be set to 0 again. To generate a unique id for each object, you can move the @id_counter outside the initialize method as a class variable.
- [ ] There also seems to be a typo in the default value for the `name` parameter so kindly make sure to fix it.